spec/models/annotation_spec.rb

```diff
@@ -38,7 +38,7 @@
 
     it 'correctly updates the mark when destroyed' do
       result.annotations.find_by(annotation_text: annotation_text).destroy
-      expect(mark.mark).to eq(3.0)
+      expect(mark.mark).to eq(nil)
```

Comment:
Good, but also add a second test when there are *multiple* deductive applied to the same result for the same criterion, and one of them is deleted. (In this case the updated mark should be based on the remaining deduction.)